service/gcs/core/gcs/gcs.go

```diff
@@ -77,6 +77,7 @@ type containerCacheEntry struct {
 	ExitHooks          []func(oslayer.ProcessExitState)
 	MappedVirtualDisks map[uint8]prot.MappedVirtualDisk
 	NetworkAdapters    []prot.NetworkAdapter
+	container          runtime.Container
 }
```

Comment:
Unfortunately this doesn't work because container is nil until the first process is created.